src/renderer/hooks/useNodesMenu.tsx

```diff
@@ -1,18 +1,29 @@
 import { CloseIcon, CopyIcon, DeleteIcon } from '@chakra-ui/icons';
-import { MenuItem, MenuList } from '@chakra-ui/react';
-import { Node } from 'reactflow';
+import { MenuDivider, MenuItem, MenuList } from '@chakra-ui/react';
+import { Node, useReactFlow } from 'reactflow';
 import { useContext } from 'use-context-selector';
-import { NodeData } from '../../common/common-types';
+import { EdgeData, NodeData } from '../../common/common-types';
 import { GlobalContext } from '../contexts/GlobalNodeState';
+import { copyToClipboard } from '../helpers/copyAndPaste';
 import { UseContextMenu, useContextMenu } from './useContextMenu';
 
 export const useNodesMenu = (nodes: Node<NodeData>[]): UseContextMenu => {
     const { removeNodesById, clearNodes, duplicateNodes } = useContext(GlobalContext);
 
     const nodeIds = nodes.map((n) => n.id);
 
+    const { getNodes, getEdges } = useReactFlow<NodeData, EdgeData>();
+
     return useContextMenu(() => (
         <MenuList className="nodrag">
+            <MenuItem
+                icon={<CopyIcon />}
+                onClick={() => {
+                    copyToClipboard(getNodes(), getEdges());
```

Comment:
This context menu is for an area selection of nodes. Why are you copying the whole chain when this context menu explicitly refers to a selected set of nodes?

![image](https://github.com/chaiNNer-org/chaiNNer/assets/20878432/69c255c7-4142-4531-8d5b-46b3884bf520)

Reply:
Read my above comment and also take a look at the copy to clipboard function.